packages/sdk/typescript/human-protocol-sdk/src/kvstore.ts

```diff
@@ -370,4 +373,40 @@ export class KVStoreClient extends BaseEthersClient {
 
     return url;
   }
+
+  /**
+   * Gets the public key of the given entity, and verify its hash.
+   *
+   * @param {string} address Address from which to get the public key.
+   * @returns {string} Public key for the given address if exists, and the content is valid
+   *
+   *
+   * **Code example**
+   *
+   * ```ts
+   * import { providers } from 'ethers';
+   * import { KVStoreClient } from '@human-protocol/sdk';
+   *
+   * const rpcUrl = 'YOUR_RPC_URL';
+   *
+   * const provider = new providers.JsonRpcProvider(rpcUrl);
+   * const kvstoreClient = await KVStoreClient.build(provider);
+   *
+   * const publicKey = await kvstoreClient.getPublicKey('0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266');
+   * ```
+   */
+  public async getPublicKey(address: string): Promise<string> {
+    const publicKeyUrl = await this.getFileUrlAndVerifyHash(
+      address,
+      KVStoreKeys.publicKey
+    );
+
+    if (publicKeyUrl === '') {
+      return '';
+    }
+
+    const publicKey = await fetch(publicKeyUrl).then((res) => res.text());
```

Comment:
can you explain this? as far as I understand getFileUrlAndVerifyHash returns the url

Reply:
Yeah, it returns URL. I think this getPublicKey function should return the content instead?

Reply:
I mean line 408. why you do this fetch?
Can't you do this?
`return publicKeyUrl;`

Reply:
Sorry, I understand now